contracts/src/v0.8/automation/dev/MercuryRegistry.sol

```diff
@@ -0,0 +1,312 @@
+|     feeds that the registry can store is unbounded.                                                                 |
+|                                                                                                                     |
+| Key Contracts:                                                                                                      |
+|   - `MercuryRegistry.sol` - stores price feed data and implements core logic.                                       |
+|   - `MercuryRegistryBatchUpkeep.sol` - enables batching for the registry.                                           |
+|   - `MercuryRegistry.t.sol` - contains foundry tests to demonstrate various flows.                                  |
+|                                                                                                                     |
+| NOTE: This contract uses Mercury v0.2. Automation will likely upgrade to v0.3 eventually, which may change some     |
+| components such as the Report struct, verification, and the StreamsLookup revert.                                   |
+|                                                                                                                     |
+| TODO:                                                                                                               |
+|   - Optimize gas consumption.                                                                                       |
+-+---------------------------------------------------------------------------------------------------------------------*/
+contract MercuryRegistry is ConfirmedOwner, AutomationCompatibleInterface, StreamsLookupCompatibleInterface {
+  error DuplicateFeed(string feedId);
+  error FeedNotActive(string feedId);
+  error StaleReport(string feedId, uint32 currentTimestamp, uint32 incomingTimestamp);
+  error InvalidFeeds();
+
+  // Feed object used for storing feed data.
+  // not included but contained in reports:
+  // - blocknumberUpperBound
+  // - upperBlockhash
+  // - blocknumberLowerBound
+  // - currentBlockTimestamp
+  struct Feed {
+    uint32 observationsTimestamp; // the timestamp of the most recent data assigned to this feed
+    int192 price; // the current price of the feed
+    int192 bid; // the current bid price of the feed
+    int192 ask; // the current ask price of the feed
+    string feedName; // the name of the feed
+    string feedId; // the id of the feed (hex encoded)
+    bool active; // true if the feed is being actively updated, otherwise false
+    int192 deviationPercentagePPM; // acceptable deviation threshold - 1.5% = 15_000, 100% = 1_000_000, etc..
+    uint32 stalenessSeconds; // acceptable staleness threshold - 60 = 1 minute, 300 = 5 minutes, etc..
+  }
+
+  // Report object obtained from off-chain Mercury server.
+  struct Report {
+    bytes32 feedId; // the feed Id of the report
```

Comment:
how is `feedId` calculated? Is it a mix of prefix version and truncated hash? If yes hash of what data.